### PR TITLE
feat(console): display topbar shadow while scrolling

### PR DESCRIPTION
**Pull request by xiaoyijun**
## Summary

feat(console): display topbar shadow while scrolling

## Testing

feat-scroll-shadow

### CODE REVIEW

File: packages/console/src/components/AppContent/index.tsx [hunk 1]
```diff
@@ -23,6 +25,8 @@ const AppContent = () => {
   const location = useLocation();
   const navigate = useNavigate();
   const { firstItem } = useSidebarMenuItems();
+  const mainRef = React.useRef<HTMLDivElement>(null);
```
Comment (demonzoo) [inline packages/console/src/components/AppContent/index.tsx:28]: nit: remove `React.`, and import from top

File: packages/console/src/hooks/use-scroll.ts [hunk 2]
```diff
@@ -0,0 +1,34 @@
+import { useState, useEffect, useCallback } from 'react';
+
+const useScroll = (ref?: HTMLDivElement) => {
```
Comment (demonzoo) [inline packages/console/src/hooks/use-scroll.ts:3]: nit: can remove `?`, make it required

File: packages/console/src/components/AppContent/index.tsx [hunk 3]
```diff
@@ -51,10 +55,10 @@ const AppContent = () => {
 
   return (
     <div className={styles.app}>
-      <Topbar />
+      <Topbar className={conditional(scrollTop > 0 && styles.topbarShadow)} />
```
Comment (demonzoo) [inline packages/console/src/components/AppContent/index.tsx:58]: nit: We usually use `classNames` for className. So:
```tsx
<Topbar className={classNames(scrollTop && styles.topbarShadow)} />
```

**Review by demonzoo — APPROVED** [review]
Please check my nit-picking comments. thx

File: packages/console/src/components/AppContent/index.tsx (hunk 1 above)
Comment (xiaoyijun) [reply packages/console/src/components/AppContent/index.tsx]: updated.

File: packages/console/src/components/AppContent/index.tsx (hunk 3 above)
Comment (xiaoyijun) [reply packages/console/src/components/AppContent/index.tsx]: image

File: packages/console/src/hooks/use-scroll.ts (hunk 2 above)
Comment (xiaoyijun) [reply packages/console/src/hooks/use-scroll.ts]: No, we can't.
The `ref.current` from `useRef` is optional.
And we cannot use `RefObj<HTMLDivElement>` as its parameter, for we will lose track of the reference when it is mounted.

File: packages/console/src/components/AppContent/index.tsx (hunk 3 above)
Comment (xiaoyijun) [reply packages/console/src/components/AppContent/index.tsx]: updated.

File: packages/console/src/hooks/use-scroll.ts (hunk 2 above)
Comment (xiaoyijun) [reply packages/console/src/hooks/use-scroll.ts]: per offline discussion, updated with `Nullable`.

**Pull request merged**